### ISSUE TITLE
Error on decoding unknown fields

### ISSUE THREAD

**Issue by dhirschfeld** [jcrist/msgspec#208]
When provided a type, is there a way to make `decode` error if the source contains fields not defined on the specified type. i.e. have the below error
```
msgspec.json.decode(Path('data.json').read_bytes(), type=MyStruct)
```
if `data.json` contains a field not defined on the `MyStruct` struct?

**Comment by jcrist** [comment]
Not currently, but this is definitely doable (it'd be easiest to add this as a config option when defining `Struct` types). Can you comment more on your use case? When would you want this behavior?

**Comment by dhirschfeld** [comment]
> Can you comment more on your use case?

Basically to just prevent typo errors. e.g. *hypothetically* you misspell a field, the config has no effect so you try to debug your code to find why it was taking the wrong path only to find out that the field was misspelled and silently ignored rather than raising an error that the config format didn't match the schema.

I think it's very valuable to be able to *not* specify the entire schema if you're only interested in certain fields - it's much less code to write and can be quicker if the entire schema doesn't need to be parsed.

In cases where you do want to decode the entire schema and it would be an error to have either missing or additional fields it would be nice to have the option to raise on missing or additional fields. Maybe a `total=True` kwarg similar to `TypedDict`? e.g.
```
msgspec.json.decode(Path('data.json').read_bytes(), type=MyStruct, total=True)
```

**Comment by jcrist** [comment]
This has been fixed in #210.
